src/zeek/network.rs

```
@@ -296,9 +296,15 @@ impl TryFromZeekRecord for Dns {
             if ttl.eq("-") {
                 vec![0]
             } else {
-                ttl.split(',')
-                    .map(|t| t.parse::<f32>().unwrap() as i32)
-                    .collect()
+                let mut ttl_vec = Vec::new();
+                for t in ttl.split(',') {
+                    ttl_vec.push(
+                        t.parse::<f32>()?
+                            .to_i32()
+                            .context("failed to convert f32 to i32")?,
+                    );
+                }
+                ttl_vec
```

I think we can maintain the split().map().collect() iteration chain, without newly introducing a mutable `ttl_vec`.

---

```
                ttl.split(',')
                    .map(|t| t.parse::<f32>().context("failed to parse string to f32").to_i32().context("failed to convert f32 to i32"))
                    .collect::<Result<_>>()?;
```

Maybe I think this would work. I think the key is the `?` after `collect`.

---

It seems that errors from parsing or conversion are ignored in the code.
(The result of the expression is `Vec<Result<i32>>`, not Result<Vec<i32>>)

If parsing or conversion fails for one of the TTLs, the result will be `vec![Ok(), Err(), Ok(), ...]`.
If we want to raise an error based on the result of the expression, we should check it again like this:
```rust
for t in ttl_vec.iter {
    if let Err(e) = t {
        bail!("{e}");
    }
}
```
However, this approach seems quite verbose.

---

I think you are right, I am sorry for confusion.